### PR TITLE
Merge develop to master

### PR DESCRIPTION
**Pull request by andela-oakinseye**
**What does this PR do?**
This Pull request completes checkpoint 2 - string class basic functionality as well as tests.

**Description of Task to be completed?**
Ensures that all methods are available as a prototype to the string class.

**How can this be manually tested?**
Download and setup the project run `npm start`

**Any background context you want to provide?**
N/A

### CODE REVIEW

File: src/string-class.js [hunk 1]
```diff
@@ -0,0 +1,165 @@
+/* @file string-class */
+const stringClassExtensions = {
+  /**
+   * hasVowels
+   * Test to see if a string contains vowels
+   * @return {Boolean} true or false
+   */
+  hasVowels() {
+    const vowels = /[aeiou]/gi;
```
Comment (emekanwankwo) [inline src/string-class.js:9]: Nitpick: It may probably be better to return the match directly instead of declaring a whole new variable for that purpose. Don't you think so? (like: `return /[aeiou]/gi.test(this)`

File: src/string-class.js [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+    return vowels.test(this);
+  },
+
+  /**
+   * toUpper
+   * Converts a string to uppercase
+   * @return {String} uppercase equivalent of the string
```
Comment (emekanwankwo) [inline src/string-class.js]: Nitpick: You can consider using the JSDoc return statement `@returns` instead of the synonym `@return` as specified [here](http://usejsdoc.org/tags-returns.html).

File: test/string-class-test.js [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+  describe('ucFirst', () => {
+    it('should return `Coolest` for `coolest`', () => {
+      expect('coolest'.ucFirst()).to.equal('Coolest');
+    });
+    it('should return `Example` for `example`', () => {
+      expect('example'.ucFirst()).to.equal('Example');
+    });
+  });
+
+  describe('isQuestion', () => {
```
Comment (emekanwankwo) [inline test/string-class-test.js:57]: What happens when I test for the edge case: ` '???'.isQuestion()` ?

File: test/string-class-test.js [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+    });
+    it('should return 1 for `hello`', () => {
+      expect('hello'.wordCount()).to.equal(1);
+    });
+    it('should return `number` for `typeof wordCount("hello")`', () => {
+      expect(typeof 'hello'.wordCount()).to.equal('number');
+    });
+  });
+
+  describe('toCurrency', () => {
```
Comment (emekanwankwo) [inline test/string-class-test.js:100]: Great job taking the edge case into account, you can also test for other edge cases like: ` '1234567890'.toCurrency() `

File: test/string-class-test.js [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+    });
+    it('should return `one one two` for `112`', () => {
+      expect('112'.numberWords()).to.equal('one one two');
+    });
+    it('should return `five Bottles` for `5Bottles`', () => {
+      expect('5Bottles'.numberWords()).to.equal('five Bottles');
+    });
+  });
+
+  describe('isDigit', () => {
```
Comment (emekanwankwo) [inline test/string-class-test.js:198]: Great job with the edge cases, you may also consider testing for cases like ` '-1'.isDigit() `

File: test/string-class-test.js [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+    it('should return false for `112`', () => {
+      expect('112'.isDigit()).to.be.false;
+    });
+    it('should return false for `one`', () => {
+      expect('one'.isDigit()).to.be.false;
+    });
+  });
+
+
+  describe('doubleCheck', () => {
```
Comment (emekanwankwo) [inline test/string-class-test.js:219]: Does the method check for special characters too? like ` 'Halo Wars!!'.doubleCheck()`

**Comment by andela-oakinseye** [comment]
@andela-cnwankwo 
Thanks a lot for reviewing, I am now working on the feedback.
